src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -414,54 +414,6 @@ internal static bool IsTagHelperFromMangledClass(TagHelperDescriptor tagHelper)
                 return ComponentMetadata.IsMangledClass(className);
             }
 
-            // Internal for testing.
-            internal static bool TrySplitNamespaceAndType(string fullTypeName, out TextSpan @namespace, out TextSpan typeName)
-            {
-                @namespace = default;
-                typeName = default;
-
-                if (string.IsNullOrEmpty(fullTypeName))
-                {
-                    return false;
-                }
-
-                var nestingLevel = 0;
-                var splitLocation = -1;
-                for (var i = fullTypeName.Length - 1; i >= 0; i--)
-                {
-                    var c = fullTypeName[i];
-                    if (c == Type.Delimiter && nestingLevel == 0)
-                    {
-                        splitLocation = i;
-                        break;
-                    }
-                    else if (c == '>')
-                    {
-                        nestingLevel++;
-                    }
-                    else if (c == '<')
-                    {
-                        nestingLevel--;
-                    }
-                }
-
-                if (splitLocation == -1)
-                {
-                    typeName = new TextSpan(0, fullTypeName.Length);
-                    return true;
-                }
-
-                @namespace = new TextSpan(0, splitLocation);
-
-                var typeNameStartLocation = splitLocation + 1;
-                if (typeNameStartLocation < fullTypeName.Length)
-                {
-                    typeName = new TextSpan(typeNameStartLocation, fullTypeName.Length - typeNameStartLocation);
-                }
-
-                return true;
-            }
-
             // Internal for testing.
             internal static string GetTextSpanContent(TextSpan textSpan, string s)
```

Comment:
If we're pulling it out, might as well pull this out as well as it's used in conjunction with `TrySplitNamespaceAndType`